### PR TITLE
bpo-46995: Deprecate missing asyncio.Task.set_name() for third-party task implementations

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue46995

### CODE REVIEW

File: Lib/asyncio/tasks.py [hunk 1]
```diff
@@ -67,7 +67,10 @@ def _set_task_name(task, name):
         try:
             set_name = task.set_name
         except AttributeError:
-            pass
+            warnings.warn("Task.set_name() was added by Python 3.8, "
```
Comment (kumaraditya303) [inline Lib/asyncio/tasks.py]:
```suggestion
            warnings.warn("Task.set_name() was added in Python 3.8, "
```

File: Misc/NEWS.d/next/Library/2022-03-12-13-50-42.bpo-46995.2kdNDg.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Deprecate missing :meth:`asyncio.Task.set_name` for third-party task
+implementations, schedule the making it mandatory in Python 3.13.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-03-12-13-50-42.bpo-46995.2kdNDg.rst]:
```suggestion
implementations, schedule making it mandatory in Python 3.13.
```

**Review by kumaraditya303 — APPROVED** [review]
LGTM with some minor tweaks.

File: Lib/asyncio/tasks.py [hunk 3]
```diff
@@ -67,7 +67,10 @@ def _set_task_name(task, name):
         try:
             set_name = task.set_name
         except AttributeError:
-            pass
+            warnings.warn("Task.set_name() was added in Python 3.8, "
+                      "the method support will be mandatory for third-party "
+                      "task implementations since 3.13.",
+                      DeprecationWarning, stacklevel=2)
```
Comment (asvetlov) [inline Lib/asyncio/tasks.py]:
```suggestion
                      DeprecationWarning, stacklevel=3)
```

**Pull request merged**